### ISSUE TITLE
Problems Caused By An ExtJS Overlay Mask Using Firefox 3.6

### ISSUE THREAD

**Issue by garytaylor** [moxiecode/plupload#294]
I have a page in my website that uses plupload as an ExtJS component.  This works brilliantly all over the site with the exception of a certain case.
In this case, the component is used in a page rather than in a popup.  This page creates a 'Gallery', then allows a user to upload images to that gallery all in one step.  As the gallery needs to be created before images can be uploaded to it, what I do is to do an XHR request to the server , once this is complete and we have a valid response, I then tell plupload to upload the files that it has got.  I have included the code at the end of this message 

After doing some debugging, I found that because I had some global code that places an overlay over the entire page and then a 'Loading' indicator in the centre of the page. This was showing during an XHR request and was automatically hidden when finished.
After this happened and before my code even had anything to do with plupload (i.e. before I told it to upload the files) - the 'Init' event triggered.  This then caused an error as below :-
n().setFileFilters() is not a function.

What this means is that the dom object for the flash could not be found or did not have a setFileFilters function.

However, when I disabled the 'Loading' indicator and the overlay - all is fine ?

So, to work around it, I have left it disabled.

Here is my code.   You dont really need to understand ExtJS that much to read it - its similar to a prototype ajax request

```
                        Ext.Ajax.request({
                           url: form.action,
                            method: form.method,
                            params: data,
                            scope: this,
                           success: function(response, opts) {
                              //var obj = Ext.decode(response.responseText);
                               //Start the files uploading
                               var obj=Ext.util.JSON.decode(response.responseText);
                               var uploadPanel=Ext.getCmp('myUploader');
                               if(obj.results>0){
```

/\* This is the bit we are interested in - this is where I start talking to the uploader when the response is received */

```
                                   /* Delaying the upload until the XHR has finished and the overlaying shim has dissapeared */
                                   new Ext.util.DelayedTask(function(){
```

/\* This code is executed after 1 second (as a trial to see if it helped !) _/
/_ All that I am doing is making plupload send to the correct url based on the newly created gallery id _/
/_ However, the error has occured before we even get here !!  */

```
                                       var newUrl="#{request.fullpath.gsub(/\/new.*$/,'/:gallery_id/images')}".replace(/:gallery_id/,obj.public_galleries[0].id)
                                       uploadPanel.uploader.settings.url=newUrl
                                       uploadPanel.url=newUrl
                                       uploadPanel.upload();

                                   }).delay(1000);
                               }else{

                               }
                           },
                           failure: function(response, opts) {
                               m=Ext.Msg.alert('Error', 'Sorry, your gallery was not created because of an error at our end. The administrator has been informed and will aim to have this fixes as soon as possible.  We apologise for any inconvenience caused.');
                               m.setIcon(Ext.MessageBox.ERROR);

                           }
                        });
```

**Comment by jayarjo** [comment]
The problem for this and similar cases is that Flash object cannot be hidden or covered with anything since this causes it to fail or reinitialize. Let it have higher zIndex then the overlay and it will work.
